grails-app/domain/org/pih/warehouse/requisition/Requisition.groovy

```diff
@@ -401,6 +402,22 @@ class Requisition implements Comparable<Requisition>, Serializable {
         return null
     }
 
+    boolean shouldSendApprovalNotification() {
+        if (status == RequisitionStatus.PENDING_APPROVAL) {
+            // if submitted for approval, then check if destination (requesting location)
+            // has enabled notifications to the approvers (fulfilling location)
+            // and should send notification to approvers
+            return destination.supports(ActivityCode.ENABLE_FULFILLER_APPROVAL_NOTIFICATIONS)
+        } else if ([RequisitionStatus.APPROVED, RequisitionStatus.REJECTED].contains(status)) {
+            // if approved or rejected, then check if destination (requesting location)
+            // has enabled notifications and should get notification  about approval or rejection
+            return destination.supports(ActivityCode.ENABLE_REQUESTOR_APPROVAL_NOTIFICATIONS)
+        }
+
+        // by default always send approval workflow notifications (if not handled above)
+        return true
```

Comment:
Given that we're handling all of the states above it doesn't seem possible. Can you identify a scenario where this could happen? 

Also I feel like I'm missing something here. It seems like the "shouldSendApprovalNotification" method should be location-dependent? In other words, we should enable fulfiller notifications on the origin and requestor approval notifications on the requestor? So asking whether to send the approval notifications should be again the stock movement or the 

Pending Approval: send Approval notification to approvers if origin has approval notifications enabled
Approved: send Approved notification to requestors if destination has approval notifications enabled
Rejected: send Rejected notification to requestors if destination has approval notifications enabled

That way we can turn each notification type on / off for each side of the stock movement. 

Can you explain why it's better for the downstream endpoint to control the notification? I'm thinking of a scenario where, as an implementer in a distribution center, I want to disable notifications for Pending Approval because I've set up my system to allow users to manage their workflow through the dashboard. I don't want one of my requestors to then re-enable notifications because they have a different workflow.

Reply:
Actually, this is an obsolete version. You are right that it should be handled above. I am gonna change it to "false", just in case someone triggers this for the wrong status that is not handled above.